sound/soc/sof/ipc4-pcm.c

```diff
@@ -120,11 +135,18 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	}
 
 	/* update PAUSED state for all pipelines that were just triggered */
-	for (i = 0; i < data->count; i++) {
-		for (j = 0; j < pipeline_list->count; j++) {
-			pipe_widget = pipeline_list->pipe_widgets[j];
-			pipeline = pipe_widget->private;
+	for (j = 0; j < pipeline_list->count; j++) {
+		pipe_widget = pipeline_list->pipe_widgets[j];
+		pipeline = pipe_widget->private;
+
+		/*
+		 * decrement trigger_count for the pipeline even if it isn't in the data's
+		 * list of pipelines that are triggered in this callback.
+		 */
+		if (!pipeline->skip_during_fe_trigger && state == SOF_IPC4_PIPE_PAUSED)
+			pipe_widget->trigger_count--;
 
+		for (i = 0; i < data->count; i++) {
 			if (data->pipeline_ids[i] == pipe_widget->instance_id) {
```

Comment:
Not related to this PR, but I am thinking we set all pipeline_ids to the same state with `sof_ipc4_set_multi_pipeline_state`.
Why don't we set all data->pipeline_ids[i] = SOF_IPC4_PIPE_PAUSED?
Can we remove the `if (data->pipeline_ids[i] == pipe_widget->instance_id)` test? And same as line 182

Reply:
There is no link from data->pipeline_ids[] back to pipeline, you would then need to look up the pipeline by it's ID, so another double loop.